src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/InProcess/DebuggerInProcess.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.Extensibility.Testing
+{
+    [TestService]
+    internal partial class DebuggerInProcess
+    {
+        public async Task<bool> SetBreakpointAsync(string fileName, int line, int character, CancellationToken cancellationToken)
```

Comment:
Yeah, this was nice, but the attribute was quite undiscoverable (which to be fair, is true of most attribute based programming systems).

@sharwell How do you feel about convention based discovery of these? ie, type names ending in `InProcess`, in the right namespace/folder etc.? Alternatively adding `[TestService]` to `EditorInProcess` etc. rather than special casing them would have helped too.